src/main/java/com/openshift/jenkins/plugins/pipeline/OcWatch.java

```diff
@@ -11,23 +11,19 @@
 import org.jenkinsci.plugins.durabletask.Controller;
 import org.jenkinsci.plugins.durabletask.DurableTask;
 import org.jenkinsci.plugins.durabletask.WindowsBatchScript;
-import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
-import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
-import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
-import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.jenkinsci.plugins.workflow.steps.*;
 import org.kohsuke.stapler.DataBoundConstructor;
 
-import javax.inject.Inject;
-
 import java.io.InputStream;
 import java.util.List;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
 import static com.openshift.jenkins.plugins.pipeline.OcAction.exitStatusRaceConditionBugWorkaround;
 
-public class OcWatch extends AbstractStepImpl {
-    
+// Based on implementation of WaitForConditionStep
+public class OcWatch extends Step {
+
```

Comment:
Granted @csrwng 's example was with watches, but isn't it conceivable we would want to do this with OcAction as well?  In theory they could be run in parallel blocks as well, no?